azurerm/internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -156,13 +156,23 @@ func resourceComputeClusterCreate(d *pluginsdk.ResourceData, meta interface{}) e
 		return tf.ImportAsExistsError("azurerm_machine_learning_compute_cluster", *existing.ID)
 	}
 
-	computeClusterProperties := machinelearningservices.AmlCompute{
-		Properties: &machinelearningservices.AmlComputeProperties{
+	computeClusterAmlComputeProperties := machinelearningservices.AmlComputeProperties{
+		VMSize:        utils.String(d.Get("vm_size").(string)),
+		VMPriority:    machinelearningservices.VMPriority(d.Get("vm_priority").(string)),
+		ScaleSettings: expandScaleSettings(d.Get("scale_settings").([]interface{})),
+	}
+
+	if d.Get("subnet_resource_id") != nil {
+		computeClusterAmlComputeProperties = machinelearningservices.AmlComputeProperties{
```

Comment:
I'd do it like this:
```suggestion
	if d.Get("subnet_resource_id") != nil {
		computeClusterAmlComputeProperties..AmlComputeProperties.Subnet = &machinelearningservices.ResourceID{ID: utils.String(d.Get("subnet_resource_id").(string))}
```